src/cmd/linuxkit/pkglib/manifest.go

```diff
@@ -0,0 +1,295 @@
+package pkglib
+
+// manifest utilities
+
+//go:generate ./gen
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/containerd/containerd/remotes"
+	"github.com/containerd/containerd/remotes/docker"
+	auth "github.com/deislabs/oras/pkg/auth/docker"
+	"github.com/docker/distribution/reference"
+	dockertypes "github.com/docker/docker/api/types"
+	"github.com/estesp/manifest-tool/pkg/registry"
+	"github.com/estesp/manifest-tool/pkg/store"
+	"github.com/estesp/manifest-tool/pkg/types"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	log "github.com/sirupsen/logrus"
+)
+
+/*
+ EVERYTHING below here is because github.com/estesp/manifest-tool moved pushManifestList into
+ a non-exported func and passed it cli. If/when it moves back, we can get rid of all of it.
+*/
```

Comment:
if this code is derived from that code you should mention it!

Reply:
Done